test/Microsoft.AspNetCore.Mvc.IntegrationTests/ArrayModelBinderIntegrationTest.cs

```diff
@@ -327,5 +327,45 @@ private class Person
             Assert.Equal(0, modelState.ErrorCount);
             Assert.True(modelState.IsValid);
         }
+
+        private class PersonWithReadOnlyAndInitializedProperty
```

Comment:
Need integration tests for all of the collection binders to ensure they handle an `IsReadOnly` property that's actually fine. In this case, the model would be
``` c#
public class Model
{
    public Dictionary<string, SubModel[]> Property1 { get; set; }
}

public class SubModel
{
    public string Property { get; set; }
}
```
Data should generally be similar to what I described in https://github.com/aspnet/Mvc/pull/5000/files#r72312472 But at least one test (say in the dictionary integration tests) should use the `[joe].Value.Property=fred` format to ensure an outer `DictionaryModelBinder` doesn't get in the way.

Other cases include `KeyValuePair<string, IFormFile>` (yes, covered in a functional test), a `SubModel` containing a `[FromHeader]` property, `KeyValuePair<string, Dictionary<string, string>`, and something generic enough to use the base `CollectionModelBinder`. Of course feel free to use a higher level list of `KeyValuePair` or a dictionary to mix things up :smiley_cat: 

Do not add additional functional tests for these scenarios.